### ISSUE TITLE
Release 4.2.0

### ISSUE THREAD

**Issue by thomaslevesque** [FakeItEasy/FakeItEasy#1240]
Can be labelled **ready** when all other issues on this milestone are closed.

- [x] ~~run code analysis in VS in *Release* mode and address all coding and style violations (send a regular pull request which must be merged before continuing)~~
- [x] ensure completeness of draft [release notes](https://github.com/FakeItEasy/FakeItEasy/releases), including non-owner contributors, if any (move release notes forward from any pre-releases to the current release)
- [x] run `tools\prepare_release.cmd x.y.z` to:
    - change "vNext" names to appropriate x.y.z version for the next GitHub Release, milestone, and release checklist (this issue)
    - create a new draft GitHub Release, milestone, and release checklist
    - create release/x.y.z branch on upstream, branching from develop
    - submit a pull request to merge the release/x.y.z branch to master
- [x] merge the pull request, _which may be done by the PR creator, unlike a regular pull request_
- [x] tag master with the new version number and push to upstream. This will cause the build server to
    - push the nupkgs to [nuget.org](https://www.nuget.org/profiles/FakeItEasy) and
    - attach all nupkgs from the tagged build to the GitHub Release.
- [x] de-list pre-release or superseded buggy NuGet packages if present
- [x] publish the GitHub release
- [x] tweet, mentioning contributors, and post link as comment here for easy retweeting ;-)
- [x] post tweet in [Gitter](https://gitter.im/FakeItEasy/FakeItEasy)
- [x] post a link to the GitHub Release in each issue in this milestone, with thanks to contributors
- [ ] close this issue
- [ ] close the [milestone](https://github.com/FakeItEasy/FakeItEasy/milestones?state=open) for this release

**Comment by blairconrad** [comment]
Shall we release when #1257 is fixed?

**Comment by thomaslevesque** [comment]
I think we should. It's a pretty serious bug.

**Comment by blairconrad** [comment]
Done. I can release (my) tonight. Or if we prefer to release while Europe is awake, can do starting around 11h00 UTC tomorrow.

**Comment by blairconrad** [comment]
```
Get FakeItEasy 4.2.0!
Fixed: Interface method does not have an implementation exception (raised and fixed by @MSkarum)
New: Detect wrong argument constraint type at runtime
https://github.com/FakeItEasy/FakeItEasy/releases/4.2.0
```

https://twitter.com/fakeiteasyfx/status/928089306878619648

**Comment by blairconrad** [comment]
This change has been released as part of [FakeItEasy 4.2.0](https://github.com/FakeItEasy/FakeItEasy/releases/tag/4.2.0).
